Packs/CiscoFirepower/Integrations/CiscoFirepower/CiscoFirepower.py

```diff
@@ -319,8 +921,643 @@ def deploy_to_devices(self, force_deploy, ignore_warning, version, device_ids) -
+            params=params,
+        )
+
+    @pagination(api_limit=API_LIMIT, items_key_path=['items'])
+    def list_intrusion_policy(
+        self,
+        limit: int = None,
+        offset: int = None,
+        expanded_response: bool = None
+    ) -> Dict[str, Any]:
+        """
+        Retrieves a list of intrusion policies.
+
+        Args:
+            limit (int, optional): Maximum number of items to return.
+                Defaults to None.
+            offset (int, optional): Item number to start looking from.
+                Defaults to None.
+            expanded_response (bool, optional): If set to true,
+                the response displays a list of objects with additional attributes.
+                Defaults to None.
+
+        Returns:
+            Dict[str, Any]: Information about intrusion policies.
+        """
+        params = assign_params(
+            limit=limit,
+            offset=offset,
+            expanded=expanded_response,
+        )
+
+        return self._http_request(
+            method='GET',
+            url_suffix='policy/intrusionpolicies',
+            params=params,
+        )
+
+    def update_intrusion_policy(
+        self,
+        intrusion_policy_id: str,
+        name: str,
+        basepolicy_id: str,
+        description: str = None,
+        inspection_mode: str = None,
+        replicate_inspection_mode: bool = None,
+    ) -> Dict[str, Any]:
+        """
+        Modifies the Intrusion Policy associated with the specified ID.
+
+        Args:
+            intrusion_policy_id (str): Identifier for Intrusion Policy.
+            name (str): Name of the Intrusion Policy.
+            basepolicy_id (str): Unique identifier representing the base policy.
+            description (str, optional): Description of the Intrusion Policy.
+                Defaults to None.
+            inspection_mode (str, optional): Indicates the inspection mode. Can be either DETECTION or PREVENTION.
+                Only applicable for Snort 3 engine.
+                Defaults to None.
+            replicate_inspection_mode (bool, optional):
+                Flag to replicate inspection mode from snort 3 version to snort 2 version.
+                Defaults to None.
+
+        Returns:
+            Dict[str, Any]: Updated Intrusion Policy information.
+        """
+        params = assign_params(
+            replicateInspectionMode=replicate_inspection_mode
+        )
+        body: Dict[str, Any] = remove_empty_elements({
+            'id': intrusion_policy_id,
+            'name': name,
+            'description': description,
+            'inspection_mode': inspection_mode,
+            'basePolicy': {
+                'id': basepolicy_id
+            }
+        })
+
+        return self._http_request(
+            method='PUT',
+            url_suffix=f'policy/intrusionpolicies/{intrusion_policy_id}',
+            params=params,
+            json_data=body,
+            timeout=600
```

Comment:
is the container configured to stay up this long? Please make sure the command (and other commands with timeout >300s) have `execution-timeout` configured in the yml.

Reply:
From what I saw in other integrations, I've implemented it this way:
Added `execution-timeout` to commands descriptions and removed `timeout` from their requests.

Did I do it right?

Reply:
hmm doesn't removing the timeout from the request make it use the default=30s? I think we should explicitly pass `timeout=600`

Reply:
I removed it because I noticed there is a global argument in XSOAR called `execution-timeout` to control this situation.
I figured this out while going through some other `Packs`.

Here is an image of its use:
![image](https://user-images.githubusercontent.com/106475467/211244248-51680f27-c573-4224-a49c-945f9f17723f.png)

And an image of the command description to notice the user to use that argument, if needed:
![image](https://user-images.githubusercontent.com/106475467/211244379-7d297c41-1e5e-4cf5-be25-477b442579cf.png)

If a new argument is needed instead of this as you suggested `timeout`, I'll add it immediately.

Reply:
Yeah, the built-in is what we need, no need to declare it. The only thing I'd change is `requests`' `timeout` value to 600, so when `execution-timeout` is used, `requests` won't limit the time to just 30 seconds (the default iirc)